### PR TITLE
Update Configuration when master conf changed

### PR DESCRIPTION
**Pull request by maobaolong**
### What changes are proposed in this pull request?

Related to #17296

### CODE REVIEW

File: core/common/src/main/java/alluxio/ClientContext.java [hunk 1]
```diff
@@ -123,8 +124,11 @@ public synchronized void loadConf(InetSocketAddress address, boolean loadCluster
     GetConfigurationPResponse response = Configuration.loadConfiguration(address,
         conf, !loadClusterConf, !loadPathConf);
     if (loadClusterConf) {
-      mClusterConf = Configuration.getClusterConf(response, conf, Scope.CLIENT);
+      InstancedConfiguration clusterConf =
+          Configuration.getClusterConf(response, conf, Scope.CLIENT);
+      mClusterConf = clusterConf;
       mClusterConfHash = response.getClusterConfigHash();
+      Configuration.updateNewConf(clusterConf);
```
Comment (dbw9580) [inline core/common/src/main/java/alluxio/ClientContext.java:131]: We should not update the static global configuration, as `ClientContext.getClusterConf` is used to get the cluster-global configuration.

File: core/common/src/main/java/alluxio/ClientContext.java [hunk 2]
```diff
@@ -123,8 +124,11 @@ public synchronized void loadConf(InetSocketAddress address, boolean loadCluster
     GetConfigurationPResponse response = Configuration.loadConfiguration(address,
         conf, !loadClusterConf, !loadPathConf);
     if (loadClusterConf) {
-      mClusterConf = Configuration.getClusterConf(response, conf, Scope.CLIENT);
+      InstancedConfiguration clusterConf =
+          Configuration.getClusterConf(response, conf, Scope.CLIENT);
```
Comment (dbw9580) [inline core/common/src/main/java/alluxio/ClientContext.java:128]: The `Scope.CLIENT` will limit the scope of configuration to that of clients' only. If we want to make sure workers can benefit from this feature then this is not sufficient.

**Comment by qian0817** [comment]
I think there are still many issues that need to be resolved here:
1. When pulling the cluster configuration from the Master node, it will be merged with the current node's configuration. However, since the configurations defined in the configuration file have a higher priority than those in the cluster configuration, it is impossible to change configurations that have already been defined in the configuration file.
2. If using the current method of cluster configuration to dynamically update Worker node configurations, it seems that only updating some of them cannot be achieved.
3. Proxy users are a security-related feature and there are security concerns when receiving their settings over a network.

File: core/common/src/main/java/alluxio/ClientContext.java (hunk 1 above)
Comment (maobaolong) [reply core/common/src/main/java/alluxio/ClientContext.java:131]: Can you help to explain why we keep cluster conf and client-side global configuration different?

File: core/common/src/main/java/alluxio/ClientContext.java (hunk 2 above)
Comment (maobaolong) [reply core/common/src/main/java/alluxio/ClientContext.java:128]: Yes, I can fix it.

File: core/common/src/main/java/alluxio/conf/Configuration.java [hunk 3]
```diff
@@ -641,4 +641,15 @@ private static Optional<Properties> loadProperties(InputStream stream) {
...
+  /**
+   * Update new configuration.
+   * @param newConf the new configuration
+   */
+  public static void updateNewConf(InstancedConfiguration newConf) {
+    InstancedConfiguration conf;
+    do {
+      conf = SERVER_CONFIG_REFERENCE.get();
+    } while (!SERVER_CONFIG_REFERENCE.compareAndSet(conf, newConf));
+  }
```
[the remark this answers is not in the document]
Comment (maobaolong) [reply core/common/src/main/java/alluxio/conf/Configuration.java:654]: done

**Comment by jiacheliu3** [comment]
I retried the failing tests and they still fail. One of them is
```
2023-05-18T02:05:08.5219898Z 02:05:08.519 [ERROR] alluxio.job.master.LostWorkerIntegrationTest.lostWorkerReregisters  Time elapsed: 15.559 s  <<< FAILURE!
2023-05-18T02:05:08.5220426Z java.lang.AssertionError
2023-05-18T02:05:08.5220872Z 	at org.junit.Assert.fail(Assert.java:87)
2023-05-18T02:05:08.5221256Z 	at org.junit.Assert.assertTrue(Assert.java:42)
2023-05-18T02:05:08.5221657Z 	at org.junit.Assert.assertTrue(Assert.java:53)
2023-05-18T02:05:08.5222347Z 	at alluxio.job.master.LostWorkerIntegrationTest.lostWorkerReregisters(LostWorkerIntegrationTest.java:70)
2023-05-18T02:05:08.5223006Z 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-05-18T02:05:08.5223655Z 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2023-05-18T02:05:08.5224305Z 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2023-05-18T02:05:08.5225139Z 	at java.lang.reflect.Method.invoke(Method.java:498)
2023-05-18T02:05:08.5225871Z 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
2023-05-18T02:05:08.5291615Z 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
2023-05-18T02:05:08.5292761Z 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
2023-05-18T02:05:08.5293432Z 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
2023-05-18T02:05:08.5294224Z 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
2023-05-18T02:05:08.5294871Z 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
2023-05-18T02:05:08.5295484Z 	at alluxio.testutils.LocalAlluxioClusterResource$1.evaluate(LocalAlluxioClusterResource.java:202)
2023-05-18T02:05:08.5296530Z 	at alluxio.heartbeat.ManuallyScheduleHeartbeat$1.evaluate(ManuallyScheduleHeartbeat.java:81)
2023-05-18T02:05:08.5297279Z 	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:288)
2023-05-18T02:05:08.5297978Z 	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:282)
2023-05-18T02:05:08.5298866Z 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
2023-05-18T02:05:08.5300059Z 	at java.lang.Thread.run(Thread.java:748)
```

@maobaolong could you take a look? Thanks!

**Pull request closed without merging**